k-distribution/k-tutorial/1_basic/02_basics/README.md

```diff
@@ -30,23 +28,43 @@ module LESSON-02-A
 endmodule
 ```
 
-Save this file and then run:
+Note that the name of the module and name of the file match. While the name of
+the module has to be written in capitals, the file name can have any 
+capitalization. It is not a requirement that the file carries the module
+name, but merely a convention. If you choose to name your file differenly, you will
+need to compile it with attribute `--main-module` followed by the name of
+the module, to point the compiler to the _main_ module of the definition. More 
+about naming conventions and main modules will follow in 
+[Lesson 1.5](../05_modules/README.md).
+
+Now, save the file and, from within the directory where you saved it, run:
 
 ```
 kompile lesson-02-a.k
 ```
 
-`kompile` is K's compiler. By default, it takes a program or specification
-written in K and compiles it into an interpreter for that input. Right now we
-are compiling a single file. A set of K files that are compiled together are
-called a **K definition**. We will cover multiple file K definitions later on.
-`kompile` will output a directory containing everything needed to execute
-programs and perform proofs using that definition. In this case, `kompile` will
-(by default) create the directory `lesson-02-a-kompiled` under the current
-directory.
+`kompile` is K's compiler. It takes a **K definition** and compiles it into an
+interpreter. Right now, you are compiling a single-file K definition, but 
+we'll cover multiple-file K definitions later on.
+
+You should get the following output:
+
+```
+[Warning] Compiler: Could not find main syntax module with name
+LESSON-02-A-SYNTAX in definition.  Use --syntax-module to specify one. Using
+LESSON-02-A as default.
+```
+
+It's just a warning, and you can ignore it for now. Compilation succeeded.
 
-Now, save the following input file in your editor as `banana.color` in the same
-directory as `lesson-02-a.k`:
+`kompile` will also output a directory containing everything needed to execute
+programs and perform proofs using the definition given as input.
+Check your working directory and you'll see directory `lesson-02-a-kompiled`
+has beed created inside. If you're curious, have a look at its contents and 
+inspect the generated documentation. 
```

Comment:
You can say that these files not only contain all the basic tools produced by a K definition, but also contain all sorts of files that can be used as input to execute these tools. Or to tell the user how its program was parsed by the compiler frontend.

If you want to highlight one of them, I would elaborate on the `definition.kore`, which is the input for the K compiler's backend. KORE is the Intermediate Representation of K and has its own definition ([link](https://github.com/runtimeverification/haskell-backend/blob/master/docs/kore-syntax.md)), and it was built to be a high-level implementation of matching logic (the Wikipedia page link should be fine as we have recently updated it).